### ISSUE TITLE
Atom freezes.

### ISSUE THREAD

**Issue by ghost** [AtomLinter/linter-sass-lint#98]
- Atom version: 1.8.0
- linter-sass-lint version: 1.5.0
- Global sass-lint version (if used): 
- Platform:
  - [ ] OSX
  - [X] Windows
  - [ ] Linux
- Expected behaviour: keep working
- Actual behaviour / issue description: I created a file _keyframe.scss with this code:

```
@mixin keyframes () {

}
```

after trying do add a dollar sing inside the parenthesis the IDE just freeze.

**Comment by DanPurdy** [comment]
This is almost identical to this issue #27

**Comment by DanPurdy** [comment]
This was also related to this issue https://github.com/tonyganch/gonzales-pe/issues/189 in the AST that sass-lint uses. Not much we can do until that gets fixed I'm afraid.

**Comment by mikemurray** [comment]
Can this issue be resolved now as tonyganch/gonzales-pe#189 seems to have been closed for a while now?

**Comment by emvidi** [comment]
Hi I am having the same issue, but I thought is because of the exclamation mark inside the mixin. The dollar sign was used used as well but it freezes as soon as I type the exclamation mark.
I am on Ubuntu 16.04 Atom 1.14.4.

**Comment by DanPurdy** [comment]
@emvidi can you paste the exact piece of code in it's smallest form that makes this fail still thanks.

**Comment by emvidi** [comment]
@DanPurdy  I don't remember the exact code now, but as soon as this happen again I will paste the code here.

**Comment by jhobz** [comment]
I have a freezing issue when trying to change
```
&:nth-child(1 ) {
    // content
}
```
to

```
&:nth-child( 1 ) {
    // content
}
```
to appease the `space-between-parens` rule I have enabled. Almost immediately after inserting the space (but only *before* the number, not after), the editor locks up. It also locks up if I add the space via another editor and then open it in Atom. I have confirmed it is a problem with this package by disabling it and not experiencing the crash.

Seems similar to this issue, so I didn't want to open a new one, but I can if you prefer.

**Comment by DanPurdy** [comment]
@jhobz the sass-lint disabled comments are enacted AFTER the AST has parsed the file. There's no way around this unfortunately as we need to have parsed the file to know that those comments even exist. Your best bet is to open an issue with [Gonzales-Pe](https://github.com/tonyganch/gonzales-pe) or contribute to that repo as I will not be able to fix these issues in the atom plugin or sass-lint.

**Comment by DanPurdy** [comment]
Both of the above issues are now confirmed fixed as sass-lint has recently moved up to using the recently released v4.2.2 of gonzales-pe. LOTS of fatal parsing errors have been handled in this update but please do keep reporting any more you come across.
